kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -222,7 +223,9 @@ public T getMandatory() {
     if (item != null) {
       return item;
     }
-    throw new KubernetesClientException("Resource not found!", 404, new StatusBuilder().withCode(404).withMessage("Not found.").build());
```

Comment:
What I mean is if we change this round so that `get` calls `getMandatory` then we can return the proper error from Kubernetes as we'll have the status from response body.